src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -302,70 +573,71 @@ static Option firstVariadicArg(int index) {
-         * <p>
          * Execution state is captured by a downcall method handle on invocation, by writing it
          * to a native segment provided by the user to the downcall method handle.
-         * For this purpose, a downcall method handle linked with the {@link #captureCallState(String[])}
+         * For this purpose, a downcall method handle linked with this
          * option will feature an additional {@link MemorySegment} parameter directly
          * following the target address, and optional {@link SegmentAllocator} parameters.
-         * This parameter represents the native segment into which the captured state is written.
+         * This parameter, called the 'capture state segment', represents the native segment into which
+         * the captured state is written.
          * <p>
-         * The native segment should have the layout {@linkplain CaptureCallState#layout associated}
-         * with the particular {@code CaptureCallState} instance used to link the downcall handle.
+         * The capture state segment should have the layout returned by {@linkplain #captureStateLayout}.
+         * This layout is a struct layout which has a named field for each captured value.
          * <p>
-         * Captured state can be retrieved from this native segment by constructing var handles
-         * from the {@linkplain #layout layout} associated with the {@code CaptureCallState} instance.
+         * Captured state can be retrieved from the capture state segment by constructing var handles
+         * from the {@linkplain #captureStateLayout capture state layout}.
          * <p>
          * The following example demonstrates the use of this linker option:
          * {@snippet lang = "java":
          * MemorySegment targetAddress = ...
-         * CaptureCallState ccs = Linker.Option.captureCallState("errno");
+         * Linker.Option ccs = Linker.Option.captureCallState("errno");
          * MethodHandle handle = Linker.nativeLinker().downcallHandle(targetAddress, FunctionDescriptor.ofVoid(), ccs);
          *
-         * VarHandle errnoHandle = ccs.layout().varHandle(PathElement.groupElement("errno"));
-         * try (Arena arena = Arena.openConfined()) {
-         *     MemorySegment capturedState = arena.allocate(ccs.layout());
+         * StructLayout capturedStateLayout = Linker.Option.capturedStateLayout();
+         * VarHandle errnoHandle = capturedStateLayout.varHandle(PathElement.groupElement("errno"));
+         * try (Arena arena = Arena.ofConfined()) {
+         *     MemorySegment capturedState = arena.allocate(capturedStateLayout);
          *     handle.invoke(capturedState);
          *     int errno = errnoHandle.get(capturedState);
          *     // use errno
          * }
          * }
+         *
+         * @param capturedState the names of the values to save.
+         * @see #captureStateLayout()
+         */
+        static Option captureCallState(String... capturedState) {
```

Comment:
What a name is not recognized? Throw IAE?

Reply:
Added